src/cryptography/hazmat/primitives/ciphers/aead.py

```diff
@@ -12,7 +12,7 @@
 
 class ChaCha20Poly1305(object):
     def __init__(self, key):
-        if not backend.chacha20poly1305_supported():
+        if not backend.aead_cipher_supported(self.__class__):
```

Comment:
Sure, what's the advantage for future reference? Beyond the general "any time you don't directly reference a dunder that's a win"?